### PR TITLE
[feature] - done runtime-dependencies separation properly:

### PR DESCRIPTION
**Pull request by valerysntx**
- nuspec fixes for proper DlibDotNet dependency on platform-specific DlibDotNet.Native runtime
 - fixes for properly restoring native dll runtime for 'net472' target framework
 - made 19.15.0.20181008-rc1
 - [WIP] `DlibDotNet.Native` nuget package lacks of linux and osx binaries, .nuspecs has needed parts -
 - win-x64 without CUDA
  
@takuya-takeuchi: consider version changes made here `19.15.0.20181008-rc1`  - can be nuget published   
 - finalizing the #60 splitting concept
-  fixes made for #61  
```
 <PackageReference Include="DlibDotNet" Version="19.15.0.20181008-rc1" />
```
@takuya-takeuchi, please review, - i'm finishing per #60 intent, will made next PR with linux platform specifics.
 Best regards!

### CODE REVIEW

**Comment by valerysntx** [comment]
For period of testing / while finishing #60 for all platforms, packages will be published on 

**https://www.myget.org/feed/nuget-dotnet/package/nuget/DlibDotNet.Native**
Do not forget to add the **rc1** tag
```
dotnet add package DlibDotNet --version 19.15.0.20181008-rc1 --source https://www.myget.org/F/nuget-dotnet/api/v3/index.json

dotnet add package DlibDotNet.Native --version 19.15.0.20181008-rc1 --source https://www.myget.org/F/nuget-dotnet/api/v3/index.json

```

**Comment by valerysntx** [comment]
image

linux-x64 done...

**Comment by valerysntx** [comment]
Briefly:
* Issues / Related : #50, #51, #59, #60, #61, #65 
- targets: win-x64, linux-x64

```<PackageReference Include="DlibDotNet" Version="19.16.99.20190105-rc1" />``` 
 
https://www.myget.org/feed/nuget-dotnet/package/nuget/DlibDotNet

**Comment by takuya-takeuchi** [comment]
Thank you very much.

I tried your nuget repositry and I found DlibDotNet.Native.dll and DlibDotNet.Native.Dnn.dll are not copied to bin directory for .NET Framework console app.
But they are copied for .NET Core apps.

We should compromise about .NET Framework?

File: nuget/build/DlibDotNet.Native.targets [hunk 1]
```diff
@@ -1,30 +1,71 @@
...
+    <Content Include="$(MSBuildThisFileDirectory)..\runtimes\osx-x64\native\DlibDotNet.Native.Dnn.dylib">
+        <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+        <Link>DlibDotNet.Native.Dnn.dylib</Link>
+    	<DeploymentContent>True</DeploymentContent>
+    </Content>
+ 	 <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\runtimes\osx-x64\native\DlibDotNet.Native.Dnn.dylib" />
    </ItemGroup>
+
+
+  <Target Name="DlibDotNetNativeRedist" Condition="'$(TargetFramework)'!=''" AfterTargets="AfterBuild">
```
Comment (valerysntx) [inline nuget/build/DlibDotNet.Native.targets]: here is the fix: 

> Nuget do not copy DlibDotNet.Native libraries to bin directory #61

**Comment by valerysntx** [comment]
Sorry, forgot to push :) 
Please check the test/DlibDotNet.Native.Tests/PackageReference.csproj 
https://github.com/takuya-takeuchi/DlibDotNet/pull/66/files#r245478719

**Comment by takuya-takeuchi** [comment]
One question.
Should I add **-p:Configuration=WINDOWS** to  Build.Nuspec.bat if build as Windows package?

**Comment by valerysntx** [comment]
No, just ```dotnet build -c Release``` everywhere

Sent from Mail for Windows 10

**Comment by takuya-takeuchi** [comment]
I execute the following command root of src\DlibDotNet

````
DlibDotNet\src\DlibDotNet>dotnet build -c Release . -r win-x64
````

Then, output error "'CallingConvention is not defined in NativeMethods'"
(Sorry, this message was output in Japanase and I translated to English. So it may not equal to your environmental outputs.)

NativeMethods.cs use preprocessor and developer should specify one we want to use.

**Comment by valerysntx** [comment]
fixed - https://github.com/takuya-takeuchi/DlibDotNet/pull/69

**Comment by takuya-takeuchi** [comment]
Thank you for fixing #69. But native libraries are not still copied to bin when using local nuget package.
Could you create nuget package and publish to your myget repository?

**Comment by valerysntx** [comment]
done: 
 https://www.myget.org/feed/nuget-dotnet/package/nuget/DlibDotNet 

`dotnet add package DlibDotNet --version 19.16.99.20190105-rc1 --source https://www.myget.org/F/nuget-dotnet/api/v3/index.json`

Could we pass checking on tests/DlibDotNet.Native.Tests ?

**Comment by takuya-takeuchi** [comment]
> Could we pass checking on tests/DlibDotNet.Native.Tests ?

Failed. I doubt test project is created dotnet core 3.0. My dev env is still 2.1.

````
> dotnet --version
2.1.200
````

But 2.1 could works file on test project. It is weird.

> https://www.myget.org/feed/nuget-dotnet/package/nuget/DlibDotNet

Package looks like to be still old. myget can support override same version by newer binary?

**Pull request merged**